1-Draft/RFCNNNN-Implicit-Line-Continuance.md

```diff
@@ -0,0 +1,229 @@
+
+```PowerShell
+New-ADUser -Name 'Jack Robinson' -GivenName 'Jack' -Surname 'Robinson' -SamAccountName 'J.Robinson' -UserPrincipalName 'J.Robinson@enterprise.com' -Path 'OU=Users,DC=enterprise,DC=com' -AccountPassword (Read-Host -AsSecureString 'Input Password') -Enabled $true
+```
+
+By itself it's not too much to handle, but in a script commands with many parameters like this can be difficult to manage. To wrap this command across multiple lines, users can either use backticks, or they can use splatting. The former is a syntactical nuisance which should really only be used in situations when PowerShell cannot implicitly intuit how lines are wrapped. The latter is helpful, but users lose the benefits of tab completion and Intellisense for parameters when they use splatting. As a workaround, they can work out the parameters they want to use for the command first, and then convert it into a splatted command, but that's onerous. Even though Visual Studio Code has an extension that makes splatting easier, as can be seen [here](https://sqldbawithabeard.com/2018/03/11/easily-splatting-powershell-with-vs-code/), once you've converted to splatting you still lose Intellisense for future updates unless you work from the command first and then add to your splatted collection, and that's just in Visual Studio Code. Other editors may or may not support that functionality, and users working in a standalone terminal won't have that available to them either.
+
+Instead, why not allow users to do this by supporting implicit line continuance when using named parameters:
+
+```PowerShell
+New-ADUser
+    -Name 'Jack Robinson'
+    -GivenName 'Jack'
+    -Surname 'Robinson'
+    -SamAccountName 'J.Robinson'
+    -UserPrincipalName 'J.Robinson@enterprise.com'
+    -Path 'OU=Users,DC=enterprise,DC=com'
+    -AccountPassword (Read-Host -AsSecureString 'Input Password')
+    -Enabled $true
+```
+
+Further, if they have some parameters they want to splat in (because splatting is not just about shortening lines -- it's very useful for users to apply common parameters/values to multiple commands), let them do this as well:
+
+```PowerShell
+$commonParams = @{
+    Path = 'OU=Users,DC=enterprise,DC=com'
+    Enabled = $true
+}
+New-ADUser
+    -Name 'Jack Robinson'
+    -GivenName 'Jack'
+    -Surname 'Robinson'
+    -SamAccountName 'J.Robinson'
+    -UserPrincipalName 'J.Robinson@enterprise.com'
+    -AccountPassword (Read-Host -AsSecureString 'Input Password')
+    @commonParams
+```
+
+And lastly, if users have a bunch of arguments they want to pass after the stop parsing sigil (`--%`), let them do this also (thanks @Jaykul for the suggestion):
+
+```PowerShell
+&"./plink.exe"
+    --% $Hostname -l $Username -pw $Password $Command
+```
+
+For the last one, `--%` stops the parsing of that command, so no further implicit line continuance may be used because that would require the parser which the scripter opted out of using for that command.
```

Comment:
I'm not sure this makes sense to me. The `--%` only ends parsing for the rest of the line (or up to a `|` pipe?), right? Shouldn't the next line be able to _continue_ the same way?

Reply:
Yes, I need to update that text.

Reply:
Yes, I need to update that text.

Reply:
Fixed. Thanks for calling that out.